### PR TITLE
ClusterLoader2:  NetworkProgrammingLatency disable on scraping kube-proxy disabled

### PR DESCRIPTION
**Pull request by RonakKhandelwal**
Fix for 
https://github.com/kubernetes/perf-tests/issues/807

### CODE REVIEW

**Comment by RonakKhandelwal** [comment]
CLA signature done.

File: clusterloader2/pkg/measurement/interface.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ type MeasurementConfig struct {
 	// TemplateProvider provides templated objects.
 	TemplateProvider *config.TemplateProvider
 	// Identifier identifies this instance of measurement.
+	ClusterLoaderConfig *config.ClusterLoaderConfig
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/interface.go]: Something is broken with the comments - comment about identifier should be moved above identifier, comment for this field should be moved above this field.

File: clusterloader2/pkg/measurement/common/slos/network_programming.go [hunk 2]
```diff
@@ -48,6 +48,10 @@ func init() {
 type netProgGatherer struct{}
 
 func (n *netProgGatherer) IsEnabled(config *measurement.MeasurementConfig) bool {
+	//NetworkProgrammingLatency measurement should detect if scraping kube-proxy is disabled and disable itself if so
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/slos/network_programming.go]: // Disable NetworkProgrammingLatency measurement if scraping kube-proxy is disabled.

**Comment by RonakKhandelwal** [comment]
Ok I'll make the changes and submit a new PR.

**Comment by wojtek-t** [comment]
> Ok I'll make the changes and submit a new PR.

You should just push to this PR - no need to open a new one.

**Comment by wojtek-t** [comment]
> Ok I'll make the changes and submit a new PR.

You should just push to this PR - no need to open a new one.

**Comment by wojtek-t** [comment]
> Ok I'll make the changes and submit a new PR.

You should just push to this PR - no need to open a new one.

**Comment by RonakKhandelwal** [comment]
I signed it

File: clusterloader2/pkg/measurement/interface.go [hunk 3]
```diff
@@ -34,6 +34,9 @@ type MeasurementConfig struct {
 	Params map[string]interface{}
 	// TemplateProvider provides templated objects.
 	TemplateProvider *config.TemplateProvider
+	//ClusterLoaderConfig instance for accessing flags
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/interface.go]: Add a space after //

File: clusterloader2/pkg/measurement/common/slos/network_programming.go [hunk 4]
```diff
@@ -48,6 +48,10 @@ func init() {
 type netProgGatherer struct{}
 
 func (n *netProgGatherer) IsEnabled(config *measurement.MeasurementConfig) bool {
+	// Disable NetworkProgrammingLatency measurement if scraping kube-proxy is disabled.
+	if nil != config && !config.ClusterLoaderConfig.PrometheusConfig.ScrapeKubeProxy {
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/slos/network_programming.go]: NIT: While checking whether a variable isn't nil is a good practice in general, you can assume that it won't be nil here, so you can remove this check.

File: clusterloader2/pkg/measurement/interface.go (hunk 3 above)
Comment (mm4tt) [reply clusterloader2/pkg/measurement/interface.go]: Also, consider removing the comment, ClusterLoaderConfig is a self-documenting name.

**Comment by RonakKhandelwal** [comment]
Sorry for the delay! The email got buried in the inbox. Let me know if any other changes are required @mm4tt

**Comment by mm4tt** [comment]
/assign @oxddr 

Janek is the creator of this measurement, assigning to him for review

**Comment by mm4tt** [comment]
I think it looks good, by there are still some issues with CLA (could you please sign it again) and gofmt fails on clusterloader2/pkg/measurement/interface.go file (you can fix it by running 
`gofmt -s -w clusterloader2/pkg/measurement/interface.go`)

**Comment by RonakKhandelwal** [comment]
Fixed and resigned. One thing that could have been the issue is on the first commit the author seems wrong? Let me know if you think I should make another pull request for the CLA to pass.

**Comment by oxddr** [comment]
> One thing that could have been the issue is on the first commit the author seems wrong?

I think that might be the issue here. What you can do is to squash all those commits into one, that you're the only author.

**Pull request merged**